src/kbmod_wf/single_chip_workflow2.py

```diff
@@ -0,0 +1,197 @@
+    from kbmod_wf.utilities.logger_utilities import get_configured_logger, ErrorLogger
+    logger = get_configured_logger("kbmod.ic_to_wu")
+
+    import numpy as np
+    from reproject.mosaicking import find_optimal_celestial_wcs
+
+    from kbmod import ImageCollection
+    from kbmod.configuration import SearchConfiguration
+    import kbmod.reprojection as reprojection
+
+    from lsst.daf.butler import Butler
+    
+    logger.info(f"Running with config {runtime_config}")
+
+    with ErrorLogger(logger):
+        # Unravell inputs
+        repo_root = runtime_config["butler_config_filepath"]
+        search_conf_path = runtime_config.get("search_config_filepath", None)
+        ic_file = inputs[0].filepath
+
+        # Run core tasks
+        logger.info("Reading ImageCollection and adjusting search limits.")
+        ic = ImageCollection.read(ic_file)
+        ic.data.sort("mjd_mid")
+        butler = Butler(repo_root)
+        search_conf = SearchConfiguration.from_file(search_conf_path)
+
+        # fixup config to match specifics of the image collection in question
+        search_conf._params["n_obs"] = len(ic)/2
+        logger.info(f"Setting search config n_obs to {search_conf._params['n_obs']}")
+        
+        # Fit the optimal WCS
+        # TODO: triple check this doesn't flip the array, I'm pretty sure it does
+        opt_wcs, shape = find_optimal_celestial_wcs(list(ic.wcs))
+        opt_wcs.array_shape = shape
+
+        wu = ic.toWorkUnit(search_config=search_conf, butler=butler, overwrite=True)
+        logger.info("Created a WorkUnit")
+
+        # we've got everything we wanted out of IC, clean it up.
+        del ic
+
+        # Resample the work unit so all pixels point to the same (ra, dec)
+        logger.info(f"Writing resampled wu to {outputs[0]}")
+        resampled_wu = reprojection.reproject_work_unit(
+            wu,
+            opt_wcs,
+            max_parallel_processes=runtime_config.get("n_workers", 8),
+        )
+        resampled_wu.to_fits(outputs[0])
+
+    return outputs
+
+
+@python_app(
+    cache=True,
+    executors=get_executors(["local_dev_testing", "gpu"]),
+    ignore_for_cache=["logging_file"],
+)
+def step2(inputs=(), outputs=(), runtime_config={}, logging_file=None):
+    from kbmod_wf.utilities.logger_utilities import get_configured_logger, ErrorLogger
+    logger = get_configured_logger("kbmod.search_task", logging_file.filepath)
+
+    import json
+    
+    from kbmod.work_unit import WorkUnit
+    from kbmod.run_search import SearchRunner
+
+    with ErrorLogger(logger):
+        wu = WorkUnit.from_fits(inputs[0].filename)
+        res = SearchRunner().run_search_from_work_unit(wu)
+        header = wu.wcs.to_header(relax=True)
+        h, w = wu.wcs.pixel_shape
+        header["NAXIS1"], header["NAXIS2"] = h, w
+        res.table.meta["wcs"] = json.dumps(dict(header))
+        res.write_table(outputs[0].filename)
```

Comment:
I think we can build the output filename from the input: `inputs[0].filename +  ".results.ecsv"`.